subprojects/core/src/main/java/org/gradle/internal/composite/CompositeBuildSettingsLoader.java

```diff
@@ -45,7 +45,10 @@ public CompositeBuildSettingsLoader(SettingsLoader delegate, ServiceRegistry bui
 
     @Override
     public SettingsInternal findAndLoadSettings(GradleInternal gradle) {
+        CompositeContextBuilder compositeContextBuilder = buildServices.get(CompositeContextBuilder.class);
```

Comment:
Is there an ordering issue with this?  Could we just inject the services we need vs the entire ServiceRegistry?